clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -910,34 +1027,47 @@ public ListOffsetResult() {
             } else if (client.hasPendingRequests(node)) {
                 log.trace("Skipping fetch for partition {} because there is an in-flight request to {}", partition, node);
             } else {
+                // TODO safe to get leader-and-epoch here? Should we use the info in partitionInfoAndEpoch?
+                ConsumerMetadata.LeaderAndEpoch leaderAndEpoch = metadata.leaderAndEpoch(partition);
```

Comment:
Should we use the leader and epoch from the fetch position?

Reply:
If we do this, then callers of `SubscriptionState#seek(TopicPartition, long)` will be missing the leader Node.